hash.c

```diff
@@ -4185,12 +4218,18 @@ hash_i(VALUE key, VALUE val, VALUE arg)
 
 /*
  *  call-seq:
- *     hsh.hash   -> integer
- *
- *  Compute a hash-code for this hash. Two hashes with the same content
- *  will have the same hash code (and will compare using <code>eql?</code>).
- *
- *  See also Object#hash.
+ *    hash.hash -> an_integer
+ *
+ *  Returns the \Integer hash value for the hash:
```

Comment:
This is something that needs a little bit more attention, I think.

The problem of calling the `class Hash` and having a `def hash` method is super confusing. The original description used `hash-code` which is also not that great because it introduces a new term... maybe we can call it "Returns the Integer digest for the contents of the hash object"

Reply:
Restoring the previous term, 'hash-code'.  I leave it for others to figure out what would be better.